swarm/storage/mru/resource.go

```diff
@@ -424,124 +358,81 @@ func (h *Handler) New(ctx context.Context, name string, frequency uint64) (stora
 		nameHash:   nameHash,
 		updated:    time.Now(),
 	}
-	h.set(nameHash.Hex(), rsrc)
+	copy(rsrc.owner[:], ownerAddr[:])
+	h.set(chunk.Addr.Hex(), rsrc)
 
 	return chunk.Addr, rsrc, nil
 }
 
-func (h *Handler) newMetaChunk(name string, startBlock uint64, frequency uint64) *storage.Chunk {
+// creates a metadata chunk
+func (h *Handler) newMetaChunk(name string, startBlock uint64, frequency uint64, ownerAddr common.Address) *storage.Chunk {
 	// the metadata chunk points to data of first blockheight + update frequency
 	// from this we know from what blockheight we should look for updates, and how often
 	// it also contains the name of the resource, so we know what resource we are working with
-	data := make([]byte, metadataChunkOffsetSize+len(name))
+	metadataChunkLength := metadataChunkOffsetSize + len(name)
+	data := make([]byte, metadataChunkLength)
 
 	// root block has first two bytes both set to 0, which distinguishes from update bytes
 	val := make([]byte, 8)
 	binary.LittleEndian.PutUint64(val, startBlock)
-	copy(data[2:10], val)
+	copy(data[:8], val)
```

Comment:
The discussion about breaking changes is here https://github.com/ethersphere/go-ethereum/pull/668#discussion_r196034862. My point is that we already have some public clusters and data on it, by changing the data encoding we are making all uploaded data invalid, which we may not care about, but we care about a numerous cryptic error messages which need investigation. If we do not care about them also, then forget that I said anything.